lib/services/serviceBus/lib/servicebusserviceclient.js

```diff
@@ -50,10 +52,26 @@ function ServiceBusServiceClient(accessKey, issuer, sharedAccessKeyName, sharedA
   }
 
   this.apiVersion = Constants.ServiceBusConstants.CURRENT_API_VERSION;
+
+  // For HTTP Keep Alive
+  this.agentOptions = {
+    keepAlive: true,
+    maxSockets: Infinity
+  };
 }
 
 util.inherits(ServiceBusServiceClient, ServiceClient);
 
+/**
+ * Sets the agent options to be sent to the request method which includes the keepAlive and maxSockets.
+ * 
+ * @param {object} options The agent options which sets the keepAlive and maxSockets properties.
+ */
+ServiceBusServiceClient.prototype.setAgentOptions = function (options) {
+  this.agentOptions.keepAlive = options.keepAlive == null ? true : options.keepAlive;
```

Comment:
do we want to validate `options` before accessing its properties?

Reply:
Do you want to check if is boolean for the keepAlive and a number for sockets, I can do that.

Reply:
No. I am thinking of checking whether `options` is `undefined` (although it would be rare that customers would call the function without passing in an argument)

Reply:
Ok, how about I fix that with `options || (options = ={});`?

Reply:
Yeah something to prevent accessing an undefined would be fine.

Reply:
Updated